src/builtins.c

```diff
@@ -378,7 +378,7 @@ JL_CALLABLE(jl_f_sizeof)
     if (jl_is_datatype(x)) {
         jl_datatype_t *dx = (jl_datatype_t*)x;
         if (dx->layout == NULL)
-            jl_error("argument is an abstract type; size is indeterminate");
+            jl_errorf("argument type %s is an abstract type; size is indeterminate", jl_symbol_name(dx->name->name));
```

Comment:
One issue here is that the argument might not be a type object at all, in which case this seems confusing.

Reply:
Hi @StefanKarpinski! This code only runs if `jl_is_datatype(x)` returns `true`. So, in which scenario would this be confusing?

Reply:
In that case, yes, it's ok to prefix with `type`. Carry on.